### ISSUE TITLE
[Tuning] GCP.K8s.PodUsingHostPIDNamespace - Typo "responce" prevents detection

### ISSUE THREAD

**Issue by zestysoft** [panther-labs/panther-analysis#1874]
## Rule Information
- **Rule ID**: GCP.K8s.PodUsingHostPIDNamespace
- **File**: `rules/gcp_k8s_rules/gcp_k8s_pod_using_host_pid_namespace.py`

## Issue Description

There is a typo on line 13 that prevents the detection from working correctly:

```python
response_host_pid = event.deep_get("protoPayload", "responce", "spec", "hostPID")
```

The key `"responce"` should be `"response"`.

## Impact

This causes `response_host_pid` to always be `None` since the misspelled path will never match any data in the event. The detection will fail to catch pods using the host PID namespace when the `hostPID` field is only present in the `protoPayload.response.spec` section (not in the request section).

## Suggested Fix

Change line 13 from:
```python
response_host_pid = event.deep_get("protoPayload", "responce", "spec", "hostPID")
```

To:
```python
response_host_pid = event.deep_get("protoPayload", "response", "spec", "hostPID")
```
